File: src/dataset/Java/house/house.java
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class House {

  private static final String firstVerse = "This is the house that Jack built.";
  private static final String lastVerse = "that lay in the house that Jack built.";

  private static final String[] items = {
      "malt",
      "rat",
      "cat",
      "dog",
      "cow with the crumpled horn",
      "maiden all forlorn",
      "man all tattered and torn",
      "priest all shaven and shorn",
      "rooster that crowed in the morn",
      "farmer sowing his corn",
      "horse and the hound and the horn"
  };

  private static final String[] actions = {
      "ate",
      "killed",
      "worried",
      "tossed",
      "milked",
      "kissed",
      "married",
      "woke",
      "kept",
      "belonged to"
  };

  String verse(int num) {
    if (num == 1) return firstVerse;
    var builder = new StringBuilder();

    builder.append(String.format("This is the %s ", items[num - 2]));

    for (int i = num - 2; i > 0; i--) {
      builder.append(String.format("that %s the %s ", actions[i - 1], items[i - 1]));
    }
    builder.append(lastVerse);
    return builder.toString();
  }

  String verses(int startVerse, int endVerse) {
    return IntStream
        .rangeClosed(startVerse, endVerse)
        .mapToObj(this::verse)
        .collect(Collectors.joining("\n"));
  }

  String sing() {
    return verses(1, 12);
  }
}

File: src/dataset/typescript/house/house.ts
const start = 'This is';
const end = ' the house that Jack built.';

type RhymeType = [left: string, right: string];

const rhymes: RhymeType[] = [
    ['the malt', 'that lay in'],
    ['the rat', 'that ate'],
    ['the cat', 'that killed'],
    ['the dog', 'that worried'],
    ['the cow with the crumpled horn', 'that tossed'],
    ['the maiden all forlorn', 'that milked'],
    ['the man all tattered and torn', 'that kissed'],
    ['the priest all shaven and shorn', 'that married'],
    ['the rooster that crowed in the morn', 'that woke'],
    ['the farmer sowing his corn', 'that kept'],
    ['the horse and the hound and the horn', 'that belonged to'],
];

export function verse(v: number): string[] {
    let result: string[] = [start];

    for (let i = v - 1; i >= 1; i--) {
        result[result.length - 1] += ` ${rhymes[i - 1][0]}`;
        result.push(`${rhymes[i - 1][1]}`);
    }

    result[result.length - 1] += end;
    return result;
}

export function verses(startVerse: number, endVerse: number): string[] {
    let result: string[] = [];

    for (let i = startVerse; i <= endVerse; i++) {
        result.push(...verse(i));
        if (i < endVerse) result.push('');
    }

    return result;
}
